lib/Sema/TypeCheckRegex.cpp

```diff
@@ -73,33 +73,29 @@ bool swift::decodeRegexCaptureTypes(ASTContext &ctx,
       if (length >= size - offset)
         return true; // Unterminated string.
       StringRef name(namePtr, length);
-      // The name is currently unused becuase we are forming a nominal
-      // `Tuple{n}` type. We will switch back to native tuples when there is
-      // variadic generics.
-      (void)name;
-      scopes.back().push_back(atomType);
+      scopes.back().push_back(
+          TupleTypeElt(atomType, ctx.getIdentifier(name)));
       offset += length + /*NUL*/ 1;
       break;
     }
     case RegexCaptureStructureCode::FormArray: {
       auto &type = scopes.back().back();
-      type = ArraySliceType::get(type);
+      type = TupleTypeElt(ArraySliceType::get(type.getType()),
+                          type.getName());
```

Comment:
Since this (and those in other `case`s) is not a `Type` anymore, I'd use something like `element`. But it's not a strong opinion.

Reply:
Suggestion applied.